main.go

```diff
@@ -247,6 +248,23 @@ var cliCommand = &cli.Command{
 						return nil
 					},
 				},
+				&cli.DurationFlag{
+					Name:        "target-time",
+					Value:       0,
+					Usage:       "the desired target time (e.g. 4m30s) for the entire test suite to complete. When 0 this flag is ignored and the test plan will not consider target time when deriving parallelism. Must be used in conjunction with --max-parallelism",
+					Destination: &cfg.TargetTime,
+					Action: func(ctx context.Context, cmd *cli.Command, v time.Duration) error {
+						if v != 0 && cmd.Int("max-parallelism") == 0 {
+							return fmt.Errorf("target-time must be used in conjunction with --max-parallelism")
+						}
+
+						if v < 0 || v > 86400*time.Second {
+							return fmt.Errorf("target-time must be between 0 and 86400 seconds")
+						}
+
+						return nil
+					},
```

Comment:
If we end up making this an env var annoyingly we'll have to move this validation to `config.Validate()`, the flag parser doesn't call these actions when it reads from environment variables.

Reply:
It seems like it was a unexpected change between v2 and v3 of the urfave/cli library.

https://github.com/urfave/cli/issues/2132
https://github.com/urfave/cli/issues/2041

There is a potential workaround suggested in the thread I haven't yet verified.

Should we go ahead and support allow this and the `max-parallelism` flag to source from environment variables?

Reply:
@gchan yeh, I looked at those tickets previously. I'd suggest moving it to `config.Validate()`. I think it's cleaner to consolidate the validation in one place.

Feel free to add an env var for `max-parallelism` at the same time. I was going to sweep back and add one anyway, it was an oversight rather than an intentional omission.